### PR TITLE
refactor: get all connections API

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What ###
- Refactored the GET API ```orgs/{orgId}/connections```
1. Implemented a well-defined interface by replacing previous existing ```Promise<object>```.
2. Ensured consistency of API status codes.
3. Modified the NATS function.
4. Tested with both positive and negative scenarios.

### Why ###
-  The implementation of a well-defined interface enhances code readability and makes the structure of the API response more explicit.
-  Ensuring consistent API status codes improves communication between different components of the system and enhances overall reliability.

### How ###

-  Replaced the return type with a well-defined interface to enhance code clarity.
-  Checked and standardized API status codes to ensure a uniform response.
- Modified the NATS function to accommodate the adjustments made in the API.

### Extra Points ###
- Exception handling is remaining for GET API /users/profile.

### CODE REVIEW

File: apps/api-gateway/src/connection/connection.controller.ts [hunk 1]
```diff
@@ -77,39 +77,44 @@ export class ConnectionController {
...
       })
-    @ApiResponse({ status: 200, description: 'Success', type: AuthTokenResponse })
+    @ApiResponse({ status: HttpStatus.OK, description: 'Success', type: AuthTokenResponse })
     async getConnections(
         @Query() getAllConnectionsDto: GetAllConnectionsDto,
         @User() user: IUserRequest,
         @Param('orgId') orgId: string,
         @Res() res: Response
-    ): Promise<Response> {
+    ): Promise<IResponseType> {
```
Comment (vivekayanworks) [inline apps/api-gateway/src/connection/connection.controller.ts:114]: This should be IResponse. Let's discuss and fix this.

File: apps/api-gateway/src/connection/connection.controller.ts [hunk 2]
```diff
@@ -77,39 +77,44 @@ export class ConnectionController {
...
 
         const { pageSize, searchByText, pageNumber, sorting, sortByValue } = getAllConnectionsDto;
-        const connectionSearchCriteria: IConnectionSearchinterface = {
+        const connectionSearchCriteria: ConnectionSearchinterface = {
```
Comment (vivekayanworks) [inline apps/api-gateway/src/connection/connection.controller.ts:117]: Let's have consistency in naming conventions for Interfaces. We have decided to use "I" as a prefix and then interface name. No need to add word "interface".

File: apps/connection/src/connection.controller.ts [hunk 3]
```diff
@@ -2,10 +2,11 @@ import { Controller } from '@nestjs/common'; // Import the common service in the
 import { ConnectionService } from './connection.service'; // Import the common service in connection module
 import { MessagePattern } from '@nestjs/microservices'; // Import the nestjs microservices package
 import {
+  ConnectionList,
+  FetchConnectionInterface,
```
Comment (vivekayanworks) [inline apps/connection/src/connection.controller.ts:6]: Let's have consistency in naming convention for Interfaces.

File: apps/api-gateway/src/connection/connection.controller.ts [hunk 4]
```diff
@@ -77,39 +77,44 @@ export class ConnectionController {
     })
     @ApiQuery({
```
Comment (vivekayanworks) [inline apps/api-gateway/src/connection/connection.controller.ts:78]: Remove ApiQuery parameters as we are handling it in DTOs.

File: apps/api-gateway/src/connection/connection.controller.ts [hunk 5]
```diff
@@ -77,39 +77,44 @@ export class ConnectionController {
...
         required: false
       })
       @ApiQuery({
         name: 'sortByValue',
+        example: 'DESC',
         type: String,
         required: false
       })
-    @ApiResponse({ status: 200, description: 'Success', type: AuthTokenResponse })
+    @ApiResponse({ status: HttpStatus.OK, description: 'Success', type: AuthTokenResponse })
```
Comment (vivekayanworks) [inline apps/api-gateway/src/connection/connection.controller.ts:108]: Replace "AuthTokenResponse" with "ApiResponseDto"

**Comment by bhavanakarwade** [comment]
@vivekayanworks Sir, there was an issue with conflicts in this branch, so I have created a PR for the same API from another branch. In it, I have also resolved the comments you had made in the previous PR.

**Pull request closed without merging**